src/api/Shrooms.DataLayer.EntityModels/Models/ApplicationUser.cs

```diff
@@ -134,9 +136,21 @@ public ApplicationUser()
 
         public int? JobPositionId { get; set; }
 
-        [ForeignKey("JobPositionId")]
+        [ForeignKey(nameof(JobPositionId))]
         public virtual JobPosition JobPosition { get; set; }
 
+        [InverseProperty(nameof(BlacklistUser.User))]
+        public virtual ICollection<BlacklistUser> BlacklistEntries { get; set; }
+
+        [NotMapped]
+        public bool UserBlacklistedMoreThanOnce
+        {
+            get
+            {
+                return BlacklistEntries.Any(entry => entry.Status != BlacklistStatus.Active);
```

Comment:
This one is a bit strange, if user was blacklisted and later the blacklist was canceled it would still say that user was blacklisted more than once, but in reality it was blacklisted once